CHANGELOG.md

```diff
@@ -3,7 +3,145 @@
 All notable changes to this project will be documented in this file. This
 project adheres to [Semantic Versioning](http://semver.org/).
 
-## Unreleased
+## 0.48.0
+
+This release rolls in security fixes from recent patch releases, along with
+a number of bugfixes, and a new builtin function.
+
+### Improved error reporting available in `opa eval`
+
+A common frustration when writing policies in OPA is when an error happens,
+causing a rule to unexpectedly return `undefined`. Using
+`--strict-builtin-errors` would allow finding the first error encountered
+during evaluation, but terminates execution immediately.
+
+To improve the debugging experience, it is now possible to display *all* of
+the errors encountered during normal evaluation of a policy, via the new
+`--show-builtin-errors` option.
+
+Consider the following error-filled policy, `multi-error.rego`:
+
+```rego
+package play
+
+this_errors(number) := result {
+        result := number / 0
+}
+
+this_errors_too(number) := result {
+        result := number / 0
+}
+
+res1 := this_errors(1)
+
+res2 := this_errors_too(1)
+```
+
+Using `--strict-builtin-errors`, we would only see the first divide by zero error:
+
+    opa eval --strict-builtin-errors -d multi-error.rego data.play
+
+```
+1 error occurred: pr-5487-ex2.rego:4: eval_builtin_error: div: divide by zero
+```
+
+Using `--show-builtin-errors` shows both divide by zero issues though:
+
+    opa eval --show-builtin-errors -d multi-error.rego data.play -f pretty
+
+```
+2 errors occurred:
+pr-5487-ex2.rego:8: eval_builtin_error: div: divide by zero
+pr-5487-ex2.rego:8: eval_builtin_error: div: divide by zero
```

Comment:
The location reported here is the same as the first error. Is this just copy-pasta in the docs or a bug in error reporting?

Reply:
It might be in how error reporting is *displayed*! :open_mouth: Observe the difference between `-f pretty` and `-f json` formats:

`-f pretty`:
```
2 errors occurred:
multi-error.rego:8: eval_builtin_error: div: divide by zero
multi-error.rego:8: eval_builtin_error: div: divide by zero
```

`-f json`:
```json
{
  "errors": [
    {
      "message": "div: divide by zero",
      "code": "eval_builtin_error",
      "location": {
        "file": "multi-error.rego",
        "row": 4,
        "col": 19
      }
    },
    {
      "message": "div: divide by zero",
      "code": "eval_builtin_error",
      "location": {
        "file": "multi-error.rego",
        "row": 8,
        "col": 19
      }
    }
  ],
  "result": [
    ...
  ]
}

```

Reply:
So with `-f pretty` users won't really know what the errors are? We should create a ticket for this if this needs to be fixed.

Reply:
https://github.com/open-policy-agent/opa/pull/5487/files#r1063723292